app/app.go

```diff
@@ -140,29 +145,43 @@ func Run() error {
-				depositHandler.RegisterDepositHandler(config.Erc721Handler, coreListener.Erc721DepositHandler)
-				depositHandler.RegisterDepositHandler(config.GenericHandler, coreListener.GenericDepositHandler)
-				depositHandler.RegisterDepositHandler(pGenericHandler, listener.PermissionlessGenericDepositHandler)
+				mh := coreExecutor.NewEVMMessageHandler(bridgeContract)
+				for _, handler := range config.Handlers {
+					switch handler.Type {
+					case "erc20":
+						{
+							depositHandler.RegisterDepositHandler(handler.Address, coreListener.Erc20DepositHandler)
+							mh.RegisterMessageHandler(handler.Address, coreExecutor.ERC20MessageHandler)
+						}
+					case "permissionedGeneric":
+						{
+							depositHandler.RegisterDepositHandler(handler.Address, coreListener.GenericDepositHandler)
+							mh.RegisterMessageHandler(handler.Address, coreExecutor.GenericMessageHandler)
+						}
+					case "permissionlessGeneric":
+						{
+							depositHandler.RegisterDepositHandler(handler.Address, listener.PermissionlessGenericDepositHandler)
+							mh.RegisterMessageHandler(handler.Address, executor.PermissionlessGenericMessageHandler)
+						}
+					case "erc721":
+						{
+							depositHandler.RegisterDepositHandler(handler.Address, coreListener.Erc721DepositHandler)
+							mh.RegisterMessageHandler(handler.Address, coreExecutor.ERC721MessageHandler)
+						}
+					}
+				}
 				depositListener := coreEvents.NewListener(client)
 				tssListener := events.NewListener(client)
 				eventHandlers := make([]coreListener.EventHandler, 0)
 				eventHandlers = append(eventHandlers, listener.NewDepositEventHandler(depositListener, depositHandler, bridgeAddress, *config.GeneralChainConfig.Id))
 				eventHandlers = append(eventHandlers, listener.NewKeygenEventHandler(tssListener, coordinator, host, communication, keyshareStore, bridgeAddress, networkTopology.Threshold))
 				eventHandlers = append(eventHandlers, listener.NewRefreshEventHandler(topologyProvider, topologyStore, tssListener, coordinator, host, communication, connectionGate, keyshareStore, bridgeAddress))
 				eventHandlers = append(eventHandlers, listener.NewRetryEventHandler(tssListener, depositHandler, bridgeAddress, *config.GeneralChainConfig.Id, config.BlockConfirmations))
-				evmListener := coreListener.NewEVMListener(client, eventHandlers, blockstore, config)
-
-				mh := coreExecutor.NewEVMMessageHandler(bridgeContract)
-				mh.RegisterMessageHandler(config.Erc20Handler, coreExecutor.ERC20MessageHandler)
-				mh.RegisterMessageHandler(config.Erc721Handler, coreExecutor.ERC721MessageHandler)
-				mh.RegisterMessageHandler(config.GenericHandler, coreExecutor.GenericMessageHandler)
-				mh.RegisterMessageHandler(pGenericHandler, executor.PermissionlessGenericMessageHandler)
+				evmListener := coreListener.NewEVMListener(client, eventHandlers, blockstore, *config.GeneralChainConfig.Id, config.BlockRetryInterval, config.BlockConfirmations, config.BlockInterval)
```

Comment:
I am wondering why do we have *uint8 type of ID in core library, looks wrong to me...

Reply:
Its uint8 on solidity as well. Why it is a pointer I have no idea and it is before my time I think.

Reply:
It's not really a problem IMO and bit of a hassle as we'll have to fix in core so I wouldn't deal with this now.